File: Java/BogoSort.java
import java.util.Random;

public class BogoSort {
    
    public static void printArray(int[] array) {
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] array = {3, 2, 5, 1, 4};
        bogoSort(array);
        System.out.print("Sorted array: ");
        printArray(array);
    }

    public static void bogoSort(int[] array) {
        Random random = new Random();
        while (!isSorted(array)) {
            shuffle(array, random);
        }
    }

    private static void shuffle(int[] array, Random random) {
        for (int i = 0; i < array.length; i++) {
            int randomIndex = random.nextInt(array.length);
            int temp = array[i];
            array[i] = array[randomIndex];
            array[randomIndex] = temp;
        }
    }

    private static boolean isSorted(int[] array) {
        for (int i = 0; i < array.length - 1; i++) {
            if (array[i] > array[i + 1]) {
                return false;
            }
        }
        return true;
    }
}

File: Kotlin/bogoSort.kt
import kotlin.random.Random

fun isSorted(arr: IntArray): Boolean {
    for (i in 0 until arr.size - 1) {
        if (arr[i] > arr[i + 1]) {
            return false
        }
    }
    return true
}

fun shuffle(arr: IntArray) {
    for (i in arr.indices) {
        val randomIndex = Random.nextInt(arr.size)
        val temp = arr[i]
        arr[i] = arr[randomIndex]
        arr[randomIndex] = temp
    }
}

fun bogoSort(arr: IntArray) {
    while (!isSorted(arr)) {
        shuffle(arr)
    }
}

fun main() {
    val arr = intArrayOf(3, 2, 5, 1, 4, 0)
    bogoSort(arr)
    println(arr.joinToString(", "))
}